src/docs/lib/shiki.ts

```diff
@@ -11,35 +11,46 @@ const theme: Theme = "github-dark";
 const defaultlangs: Lang[] = ["html", "tsx"];
 const bg: React.CSSProperties["backgroundColor"] = "#011627";
 
+const highlighterCache = new Map<string, Promise<Highlighter>>();
+
+/**
+ * Get a shiki highlighter instance.
+ * Cached by theme and languages.
+ */
 export async function getHighlighter({
   langs = defaultlangs,
 }: {
   langs?: Lang[];
-} = {}) {
-  /** Preload NO languages in development */
-  const isDevelopment = process.env.NODE_ENV === "development";
+} = {}): Promise<Highlighter> {
+  const key = [theme, ...langs].join(",");
+  if (!highlighterCache.has(key))
+    highlighterCache.set(
+      key,
+      /* ✅ Create a highlighter instance with a theme */
+      getHighlighterFromShiki({
+        theme,
+        langs,
+      })
+    );
 
-  /* ✅ Create a highlighter instance with a theme */
-  return await getHighlighterFromShiki({
-    theme,
-    langs: isDevelopment ? [] : langs,
-  });
+  const promise = highlighterCache.get(key);
+  if (!promise) throw new Error("Promise not found");
+  return await promise;
```

Comment:
What's the purpose of throwing this error?

If the key isn't found in the cache, it sets a key-value. If it is found then it should just return the promise:

```suggestion
  return await highlighterCache.get(key);
```

Did you find any issues during testing?

Reply:
Actually was not sure for while, but it was to satisfy TS (get returning Promise or undefined). Wanted to check for better solution, and while there is probably no clean one, I think in this case its okay to do as we can be smart this time:
```suggestion
  return await highlighterCache.get(key)!;
```
Is it okay from your side (I was kind of guided not to overuse `!` previously but I do not see issue with it in this place)?

Reply:
How about this?

```ts
const highlighterStore = new Map<string, Promise<Highlighter>>();

/**
 * Get a shiki highlighter instance.
 * Cached by theme and languages.
 */
export async function getHighlighter({
  langs = defaultlangs,
}: {
  langs?: Lang[];
} = {}): Promise<Highlighter> {
  const key = [theme, ...langs].join(",");
  const highlighterCache = highlighterStore.get(key);

  // If cache is found return early
  if (highlighterCache) {
    return highlighterCache;
  }

  // Create an instance of highlighter
  // Set in store and
  // Return the highlighter promise
  const highlighterPromise = getHighlighterFromShiki({
    theme,
    langs,
  });
  highlighterStore.set(
    key,
    highlighterPromise
  );
  return highlighterPromise;
}
```

Reply:
Yes, that looks quite reasonable (almost feels like I wrote this code style before 😓 ), thanks